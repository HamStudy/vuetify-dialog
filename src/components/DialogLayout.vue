<template>
  <v-dialog @keydown.esc="dismiss"
    v-model="isActive" :max-width="getWidth"
    :persistent="persistent || loading" :fullscreen="fullscreen" ref="vdialog">
    <slot/>
  </v-dialog>
</template>

<script>
export default {
  props: {
    fullscreen: Boolean
  },
  methods: {
    _destroy () {
      this.$refs.vdialog.$refs.dialog.addEventListener('transitionend', this.onTransitionEnd)
    },
    onTransitionEnd (event) {
      if (['opacity', 'z-index'].indexOf(event.propertyName) >= 0) {
        this.$refs.vdialog.$refs.dialog.removeEventListener('transitionend', this.onTransitionEnd)
        this.$destroy()
      }
    }
  }
}
</script>
