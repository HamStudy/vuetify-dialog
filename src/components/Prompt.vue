<template>
  <div>
    <v-card tile>
      <v-card-title v-if="title">
        <h3 class="headline mb-0" v-text="title" />
      </v-card-title>
      <v-container>
        <v-text-field
          autofocus
          @keypress.enter="$emit('close', editedValue)"
          v-model="editedValue"
          :label="text"
          required
        />
      </v-container>
      <v-card-actions>
        <v-spacer/>
        <v-btn v-for="button in parsedButtons" :key="button.key"
          :color="button.color || 'green'"
          @click="onSubmit(button)"
          v-html="translate(button.text)"
          flat/>
      </v-card-actions>
    </v-card>
  </div>
</template>

<script>

import ButtonsMixin from 'vuedl/src/mixins/buttons'
import ConfirmMixin from 'vuedl/src/mixins/confirm'
import ColorsMixin from '../mixins/colorable'

export default {
  layout: 'default',
  mixins: [ConfirmMixin, ColorsMixin, ButtonsMixin],
  props: {
    icon: String,
    persistent: Boolean,
    value: String
  },
  data () {
    return {
      editedValue: this.value
    }
  },
  computed: {
    getIcon () {
      return this.icon || this.type
    }
  },
  methods: {
    onSubmit (button) {
      if (button.key) {
        this.$emit('close', this.editedValue)
      } else {
        this.$emit('close', false)
      }
    },
    translate (text) {
      return (typeof this.$t === 'function') ? this.$t(text) : text
    }
  }
}
</script>
