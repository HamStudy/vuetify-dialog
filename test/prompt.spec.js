import DialogManager from 'vuedl/src/manager'
import Prompt from '../src/components/Prompt'
import DialogLayout from '../src/components/DialogLayout'
import Vue from 'vue'
import Vuetify from 'vuetify'
import { sleep, addElemWithDataAppToBody, disableTransitions } from './utils'
Vue.use(Vuetify)
const vuetify = new Vuetify()

describe('manager', () => {
  let manager
  disableTransitions()
  addElemWithDataAppToBody()

  manager = new DialogManager({ context: { vuetify, store: {} } })
  manager.layout('default', DialogLayout)
  manager.component('prompt', Prompt, {
    actions: {
      false: 'Cancel',
      true: 'OK'
    }
  })

  test('Test default prompt', async () => {
    const dialog = await manager.prompt({
      text: 'Test prompt',
      tytle: 'Title'
    })
    await sleep(100)
    expect(dialog.element).toMatchSnapshot()
    dialog.close()
    await sleep(100)
    expect(document.body).toMatchSnapshot()
  })

  test('Prompt confirm', async () => {
    let dlg = await manager.prompt({
      text: 'Test prompt',
      tytle: 'Title',
      value: 'test'
    })
    await sleep(100)
    expect(dlg.vm.$el).toMatchSnapshot()
    setTimeout(() => {
      dlg.vmd.$refs.card.trigger(true)
    }, 5)
    let res = await dlg.wait()
    expect(res).toBe('test')
    await sleep(100)
    expect(document.body).toMatchSnapshot()
  })
})
