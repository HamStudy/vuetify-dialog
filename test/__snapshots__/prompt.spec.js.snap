// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Prompt confirm 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
/>
`;

exports[`manager Test default prompt 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
/>
`;
